dangerzone/document.py

```diff
@@ -82,3 +99,27 @@ def set_default_output_filename(self) -> None:
         self.output_filename = (
             f"{os.path.splitext(self.input_filename)[0]}{SAFE_EXTENSION}"
         )
+
+    def is_unsafe(self) -> bool:
+        return self.state is Document.STATE_UNSAFE
```

Comment:
I would suggest changing this check to `self.state is not Document.STATE_SAFE`. This way, if we have a document in any other state than strictly `STATE_UNSAFE`, we won't handle it as if it's safe (i.e., double negation does not work here).

Reply:
I understand the reasoning but we need some way to distinguish unsafe (because unconverted documents) from unsafe (failed to convert) ones. This is important in the GUI multi-document support so we know which image to show:

https://github.com/freedomofpress/dangerzone/blob/c81fde6670193f3c07453cbc87a4760f71f4eb4d/dangerzone/gui/main_window.py#L665-L673

Reply:
This shouldn't be a problem:

```python
 def update_status_image(self) -> None: 
     if self.document.is_safe(): 
         self.status_image.setPixmap(self.img_status_safe) 
     elif self.document.is_converting(): 
         self.status_image.setPixmap(self.img_status_converting) 
     elif self.document.is_failed(): 
         self.status_image.setPixmap(self.img_status_failed)
     elif self.document.is_unsafe(): 
         self.status_image.setPixmap(self.img_status_unsafe)
     else:
         raise Exception("Could not handle document status: ...")
```

If we deem that "unsafe" is a more general status, and safe/converting/failed are more specific ones, then we can simply change the order.

Reply:
Ah, I see. Yes, then `unconverted` makes more sense, and is less easy to abuse it by mistake.

Reply:
OK